cpp/include/cudf/table/table_device_view.cuh

```diff
@@ -75,10 +75,7 @@ class table_device_view : public detail::table_device_view_base<column_device_vi
  public:
   static auto create(table_view source_view, cudaStream_t stream = 0)
   {
-    auto deleter = [](table_device_view* t) {
-      t->destroy();
-      delete t->_descendant_storage;
```

Comment:
`t->descendant_storage` is a raw pointer to a device_buffer. If it were not a raw pointer there wouldn't have been a leak, right?

Reply:
May be better to use unique_ptr.

Reply:
So choosing something like `std::unique_ptr` doesn't appear to work, and honestly I don't know all the reasons why and would love to know. 

I get:

```
../src/sort/is_sorted.cu(40): error: function "cudf::table_device_view::table_device_view(const cudf::table_device_view &)" (declared implicitly) cannot be referenced -- it is a deleted function
```

We delete the `default` constructor, but let the other constructors/`operator=` to be `default`. I even tried to default initialize it to `{}`. 

Also I see errors like this:

```
_deps/thrust-src/thrust/iterator/detail/zip_iterator.inl(97): error: identifier "std::unique_ptr< ::rmm::device_buffer,     ::std::default_delete< ::rmm::device_buffer> > ::~unique_ptr" is undefined in device code
```

Maybe a limitation of nvcc or thrust? But I honestly need some help explaining both cases, if someone here could clarify.

Reply:
Ok we discussed offline, the second issue (`~unique_ptr" is undefined in device code`), is expected here, because `table_device_view` needs to go to the device, and `unique_ptr` has non-device methods. 

I am still not clear on the first issue with the copy constructors.

Reply:
My fault, it has to be a raw pointer since objects of this type are passed directly to device code.